File: 26th/somilee/number_block.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>

using namespace std;

int p[50005];
vector<int> answer;
vector<int> solution(long long begin, long long end) {
	//vector<int> answer;
    long long num, div;
    for(num = begin; num <= end; num++) {
        int end_div = (int)sqrt(num);
        bool flag = true;
        for(div = 2; div <= end_div; div++) {
            if (num % div == 0) {
				if ()
                flag = false;
                break;
            }
        }
        if (flag && num != 1)
            answer.push_back(1);
        else
            answer.push_back(num/div);
    }
    return answer;
}

int main(void) {
	long long begin = 100000014;
	long long end = 100000016;
	solution(begin, end);
	// for(long long i = 1; i <= end; i++) {
	// 	cout << i << ' ';
	// }
	cout << '\n';
	for(int n : answer) {
		cout << n << ' ';
	}
}

File: 26th/jongmlee/NumberBlock.java
class NumberBlock {
    public int[] solution(long begin, long end) {
        int[] answer = new int[(int)(end - begin + 1)];
    		int num=0;
    		long total=0;
            int j;
    		for(long i = begin; i <= end; i++) {
    			int sqrt = (int)Math.sqrt(i);
    			for(j = 2; j <= sqrt; j++) {
    				if(i % j == 0) {
    					long div = i / j;
    					if(div <= 10000000) {
    						answer[num] = (int)(div);
    						break;
    					} else {
                            answer[num] = Math.max(answer[num], j);
                        }
    				}
    			}
    			if(i != 1 && answer[num] == 0) {
                    answer[num] = 1;
                }
    			total += sqrt - j + 1;
    			num++;
    		}
    		return answer;
    }
}
